plugins/module_utils/network/sonic/config/dcbx/dcbx.py

```diff
@@ -0,0 +1,440 @@
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = diff
+        requests = []
+        requests = self.get_modify_specific_dcbx_interfaces_param_requests(
+            commands)
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, 'merged')
+        else:
+            commands = []
+        return commands, requests
+
+    def get_length(self, obj):
+        if isinstance(obj, list):
+            if obj and isinstance(obj[0], dict):
+                return len(obj[0])
+            else:
+                return 0
+        elif isinstance(obj, dict):
+            return len(obj)
+        else:
+            return 0
+
+    def remove_default_entries(self, data):
+        pop_list = []
+        new_data = {}
+
+        glbl = data.get('global')
+        if glbl:
+            enabled = glbl.get('enabled')
+            if enabled is not True:
+                glbl.pop('enabled')
+
+        intfs = data.get('interfaces')
+        if intfs:
+            for intf in intfs:
+                pop_item = False
+                if 'pfc_tlv_enabled' in intf and intf.get('pfc_tlv_enabled') is True:
+                    intf.pop('pfc_tlv_enabled')
+                    pop_item = True
+                if 'ets_configuration_tlv_enabled' in intf and intf.get('ets_configuration_tlv_enabled') is True:
+                    intf.pop('ets_configuration_tlv_enabled')
+                    pop_item = True
+                if 'ets_recommendation_tlv_enabled' in intf and intf.get('ets_recommendation_tlv_enabled') is True:
+                    intf.pop('ets_recommendation_tlv_enabled')
+                    pop_item = True
+                if 'enabled' in intf and intf.get('enabled') is True:
+                    intf.pop('enabled')
+                    pop_item = True
+                if 'name' in intf and len(intf) == 1 and pop_item:
+                    idx = intfs.index(intf)
+                    pop_list.insert(0, idx)
+
+            for idx in pop_list:
+                intfs.pop(idx)
+            if not intfs:
+                data.pop('interfaces')
+
+    def _state_deleted(self, want, have, diff):
+        """ The command generator when state is deleted
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = []
+        requests = []
+
+        is_delete_all = False
+        if not want:
+            commands = deepcopy(have)
+            is_delete_all = True
+        else:
+            commands = get_diff(want, diff, TEST_KEYS)
+        self.remove_default_entries(commands)
+        commands = remove_empties(commands)
+        requests = self.get_delete_dcbx_requests(commands, have, is_delete_all)
+        if len(requests) == 0:
+            commands = []
```

Comment:
This argspec is a dict at the top level.

```suggestion
            commands = {}
```

Reply:
@kerry-meyer I checked in other files, e.g vrfs.py, they all use commands = []

Reply:
The vrfs argspec, at the top level, is a list, so that is correct for the "vrfs" case.

The DCBx argspec is a dict at the top level, so it is more correct to initialize the corresponding commands object as an empty dictionary. The initialization type should match the operations that can be performed on the object. For example, if subsequent instructions attempt to execute "commands.update()" on this object, it will cause an exception. If an attempt is made to "append" an item to an object initialized this way, it will "work", although it is incorrect and should throw an exception.

If you search on "commands = {}" in our repo, you will find it used for modules having argspecs that are dictionaries at the top level. The "empty list" initialization may be used incorrectly in some modules. This is incorrect, but it depends what happens after the initialization whether an exception will occur or not. Even for cases where no exception is currently generated as a result of this type of incorrect initialization, it is confusing to others reading the code and incorrect to initialize an object that is actually a dictionary as an empty list.

Reply:
Have to reverse back to []. Seems regression validation script has "empty []" only.

Reply:
This is okay as used here. As a default return value for the "update_states" output, an empty list is appropriate.